sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -142,13 +142,25 @@ case class RowDataSourceScanExec(
       handledFilters
     }
 
+    val optionOutputMap =
+      if (pushedDownOperators.limit.isDefined && pushedDownOperators.sortValues.nonEmpty) {
+        val pushedTopN =
+          s"""
+             |ORDER BY ${seqToString(pushedDownOperators.sortValues.map(_.describe()))}
+             |LIMIT ${pushedDownOperators.limit.get}
+             |""".stripMargin
+        Map("pushedTopN" -> pushedTopN)
+    } else {
+        pushedDownOperators.aggregation.fold(Map[String, String]()) { v =>
+          Map("PushedAggregates" -> seqToString(v.aggregateExpressions),
+            "PushedGroupByColumns" -> seqToString(v.groupByColumns))} ++
+          pushedDownOperators.limit.map(value => "PushedLimit" -> s"LIMIT $value")
+    }
+
     Map(
       "ReadSchema" -> requiredSchema.catalogString,
       "PushedFilters" -> seqToString(markedFilters.toSeq)) ++
-      pushedDownOperators.aggregation.fold(Map[String, String]()) { v =>
-        Map("PushedAggregates" -> seqToString(v.aggregateExpressions),
-          "PushedGroupByColumns" -> seqToString(v.groupByColumns))} ++
-      pushedDownOperators.limit.map(value => "PushedLimit" -> s"LIMIT $value") ++
```

Comment:
Why do we remove them? I think we should just append the top n information
```
val topNInfo = if (pushedDownOperators.limit.isDefined && pushedDownOperators.sortValues.nonEmpty) {
  Some(.. -> ...)
} else {
  None
}
...
pushedDownOperators.limit.map(value => "PushedLimit" -> s"LIMIT $value") ++
topNInfo ++
```

Reply:
Updated.